### PR TITLE
tfidf runtime enhancement changes

### PR DESCRIPTION
**Pull request by AdityaSoni19031997**
### Description

The minimal changes made to the `tf_idf_utils` file in this PR helps in reducing the overall runtime by avoiding the looping and slicing the pandas DataFrame.

The PR helps in resolving the issue raised in #1568. 

### Checklist:

- [x] I have followed the [contribution guidelines](../CONTRIBUTING.md) and code style for this project.
- [ ] I have added tests covering my contributions.
- [ ] I have updated the documentation accordingly.
- [x] This PR is being made to `staging branch` and not to `main branch`.

Looking forward to the feedback on the PR and any other changes needed.

Best,
Aditya.

### CODE REVIEW

**Comment by AdityaSoni19031997** [comment]
Can the workflow be approved so that we can see the results now?

**Comment by miguelgfierro** [comment]
@AdityaSoni19031997 approved!

File: recommenders/models/tfidf/tfidf_utils.py [hunk 1]
```diff
@@ -239,13 +241,17 @@ def __create_full_recommendation_dictionary(self, df_clean):
         # Similarity measure
         cosine_sim = linear_kernel(self.tfidf_matrix, self.tfidf_matrix)
 
+        # _sorted_idx has the indices that would sort the array.
+        _sorted_idx = np.argsort(cosine_sim, axis=1)
+
+        _data = list(df_clean[self.id_col].values)
+        _len_df_clean = len(df_clean)
```
Comment (miguelgfierro) [inline recommenders/models/tfidf/tfidf_utils.py:248]: this is a minor thing, but in the repo we normally not use the notation `_data` or `_sorted...`, would you mind to change all those entries to `data`, `sorted_idx` or `len_df_clean`

Comment (AdityaSoni19031997) [reply recommenders/models/tfidf/tfidf_utils.py:248]: Sure, thanks for the feedback. I will update them.

**Review by miguelgfierro — APPROVED** [review]
Really good contribution, thanks!

**Comment by AdityaSoni19031997** [comment]
Thanks Miguel! Looking forward to explore bits and pieces of this repository!

**Comment by AdityaSoni19031997** [comment]
cc @miguelfierro I am just curious, Why the repository doesn't use PyTorch?

**Comment by miguelgfierro** [comment]
> Why the repository doesn't use PyTorch?

hehe, good question. Parts of the code in this repo are old, Microsoft Research used to use TF, and now they are moving to PyTorch. In future releases you will probably see more code in PyTorch, but we will keep supporting TF. 

@anargyri has been having recently some fun with TF https://github.com/microsoft/recommenders/pull/1565

**Comment by AdityaSoni19031997** [comment]
Haha, Not a TF fan! I like PyTorch more than TF.

Is there any brach where people are working in porting the snips to torch?

Thanks.

**Comment by miguelgfierro** [comment]
> Is there any brach where people are working in porting the snips to torch?

we are not planning to port the current code to PyTorch, the future one will be developed in PyTorch

**Pull request merged**